dsn/src/builder.rs

```diff
@@ -166,7 +167,7 @@ pub struct DsnOptions<C, PieceByHash, SegmentHeaderByIndexes> {
     /// Keypair for networking
     pub keypair: subspace_networking::libp2p::identity::Keypair,
     /// Get piece by hash handler
-    pub get_piece_by_hash: PieceByHash,
+    pub get_piece_by_index: PieceByHash,
```

Comment:
`PieceByHash` generic should be renamed

Reply:
Will do this in next PR. Do not want to block this PR for another couple of hours.